### PR TITLE
Issue/1102 fragmentize add order note

### LINKED ISSUE #1102: Fragment-ize "Add Order Note" activity
Prior to converting to the the navigation component, we should fragment-ize some of activities. First up is the "Add Order Note" activity.

### PR DESCRIPTION
**Pull request by nbradbury**
Closes #1102 by converting the "Add Order Note" activity to a fragment. This was a necessary step before migrating to the navigation component.

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WooCommerce/src/main/res/layout/fragment_add_order_note.xml [hunk 1]
```diff
@@ -40,8 +27,33 @@
...
+        <ImageView
+            android:id="@+id/addNote_icon"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginEnd="@dimen/margin_large"
+            android:contentDescription="@string/orderdetail_addnote_contentdesc"
+            app:srcCompat="@drawable/ic_note_private"/>
     </LinearLayout>
 
+    <ProgressBar
```
Comment (anitaa1990) [inline WooCommerce/src/main/res/layout/fragment_add_order_note.xml]: Not sure why there is a progress bar here? I noticed this was present before this PR, in the `activity_add_order_note` but I found that it’s not being used anywhere so we could remove it 🤷‍♀ ?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/AddOrderNoteFragment.kt [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    override fun onResume() {
+        super.onResume()
+        AnalyticsTracker.trackViewShown(this)
+
+        activity?.let {
+            (it as AppCompatActivity).supportActionBar?.setHomeAsUpIndicator(R.drawable.ic_gridicons_cross_white_24dp)
+        }
+    }
+
+    override fun onPause() {
```
Comment (anitaa1990) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/AddOrderNoteFragment.kt]: When I click on the `Settings` popup menu and click on back button, the cross icon is not visible anymore and the toolbar title is changed as well.




I removed this line from `onPause` and tried again and it was working correctly.

**Comment by nbradbury** [comment]
Thanks for the review, @anitaa1990! I corrected the issues you found, so ready for another round. Note that I opted to hide the Settings & Help menu items, to fit previous behavior.

**Comment by nbradbury** [comment]
> I enabled `Do not keep activities` and when clicking on `Add` menu button in `AddOrderNoteFragment`, I got this crash:

Well, that was a silly mistake on my part. It was caused by expecting the `listener` to be valid, which it would not be if the activity was destroyed.

This turned out to be thornier to fix than I expected, though, since the listener was how we told the order detail to add the note. After trying a few unreliable ways to communicate between the two fragments, I decided to instead have the "Add order note" fragment take care of dispatching the request to add the order note, making the listener unnecessary (see a60b3ce).

> When `Do not keep activities` is enabled, the toolbar title is changed to `WooCommerce` when fragment is opened again...

This looks like an overall architectural issue in the app, since the same thing happens with the order detail screen. I've opened #1119 to address this separately.

> The soft keypad is still displayed when another tab is selected from AddOrderNoteFragment.

This also appears to be an architectural issue. We want to hide the keyboard when the fragment closes, which requires relying on `onDestroy` or `onDestroyView` or even `onStop`, but none of these are called when you tap another tab (they _are_ called, though, when you tap the already selected tab). I've opened #1120 to address this separately.

**Pull request merged**